src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -8488,6 +8489,76 @@ class StubGenerator: public StubCodeGenerator {
+      //     <N>      is the number of elements in the vector
+      //              "2/4" for neon, and "x" for sve
+      //     <U>      is the precision level
+      //              "u10/u05" represents 1.0/0.5 ULP error bounds
+      //               We use "u10" for all operations by default
+      //               But for those functions do not have u10 support in SLEEF, we use "u05" instead
+      //     <suffix> indicates neon/sve
+      //              "sve/advsimd" for sve/neon implementations
+      //     e.g. Sleef_sinfx_u10sve is the method for computing vector float sin using SVE instructions
+      //          Sleef_cosd2_u10advsimd is the method for computing 2 elements vector double cos using NEON instructions
+      //
+      log_info(library)("Loaded library %s, handle " INTPTR_FORMAT, JNI_LIB_PREFIX "sleef" JNI_LIB_SUFFIX, p2i(libsleef));
+
+      // Math vector stubs implemented with SVE for scalable vector size.
+      if (UseSVE > 0) {
+        for (int op = 0; op < VectorSupport::NUM_VECTOR_OP_MATH; op++) {
+          int vop = VectorSupport::VECTOR_OP_MATH_START + op;
+          // Skip "tanh", since there is performance regression
+          if (vop == VectorSupport::VECTOR_OP_TANH) {
+            continue;
+          }
+
+          // SLEEF does not support u10 level of "hypot" yet.
+          const char* ulf = (vop == VectorSupport::VECTOR_OP_HYPOT) ? "u05" : "u10";
+
+          snprintf(ebuf, sizeof(ebuf), "Sleef_%sfx_%ssve", VectorSupport::mathname[op], ulf);
+          StubRoutines::_vector_f_math[VectorSupport::VEC_SIZE_SCALABLE][op] = (address)os::dll_lookup(libsleef, ebuf);
+
+          snprintf(ebuf, sizeof(ebuf), "Sleef_%sdx_%ssve", VectorSupport::mathname[op], ulf);
+          StubRoutines::_vector_d_math[VectorSupport::VEC_SIZE_SCALABLE][op] = (address)os::dll_lookup(libsleef, ebuf);
+        }
+      }
+
+      // Math vector stubs implemented with NEON for 64/128 bits vector size.
+      for (int op = 0; op < VectorSupport::NUM_VECTOR_OP_MATH; op++) {
+        int vop = VectorSupport::VECTOR_OP_MATH_START + op;
+
+        // SLEEF does not support u10 level of "hypot" yet.
+        const char* ulf = (vop == VectorSupport::VECTOR_OP_HYPOT) ? "u05" : "u10";
+
+        snprintf(ebuf, sizeof(ebuf), "Sleef_%sf4_%sadvsimd", VectorSupport::mathname[op], ulf);
+        StubRoutines::_vector_f_math[VectorSupport::VEC_SIZE_64][op] = (address)os::dll_lookup(libsleef, ebuf);
+
+        snprintf(ebuf, sizeof(ebuf), "Sleef_%sf4_%sadvsimd", VectorSupport::mathname[op], ulf);
+        StubRoutines::_vector_f_math[VectorSupport::VEC_SIZE_128][op] = (address)os::dll_lookup(libsleef, ebuf);
+
+        snprintf(ebuf, sizeof(ebuf), "Sleef_%sd2_%sadvsimd", VectorSupport::mathname[op], ulf);
+        StubRoutines::_vector_d_math[VectorSupport::VEC_SIZE_128][op] = (address)os::dll_lookup(libsleef, ebuf);
+      }
+    } else {
+      if (FLAG_IS_DEFAULT(UseSleefLib)) {
+        log_info(library)("Fail to load sleef library!");
```

Comment:
The library name being looked up is probably useful here too.

Reply:
Thanks for the review! I will address this in next commit.

Reply:
It is disabled (i.e. `""`) by default. So the library name is not needed here.